### ISSUE TITLE
[wasm] VFS/common WasmApp properties ignored in wasmbrowser template

### ISSUE THREAD

**Issue by RedMike** [dotnet/runtime#99223]
### Description

There are a number of public properties listed in `WasmApp.Common.targets` (previously `WasmApp.targets`) like `WasmFilesToIncludeInFileSystem`. However, at least in 8.0.2, the logic does not run even for a fresh project from `wasmbrowser` template. This seems to be because some properties/items are tied to `WasmGenerateAppBundle` being true (which leads to `_WasmGenerateAppBundle` running), but that is forced to false when `_UsingBlazorOrWasmSdk` is true within the toolchain.

This means that there's no actual `AppBundle` folder created, VFS isn't exposed at all so there's no apparent way to add files to it, and some other properties listed in the targets file don't seem to have an effect.

Setting `WasmGenerateAppBundle` to true, or attempting to unset `_UsingBlazorOrWasmSdk`, in the csproj does not work because these values are being set in the toolchain file.

The documentation for `WasmGenerateAppBundle` at least seems to be wrong if this is intentional: 
```                                              
Defaults to true.
This is useful for projects that want to handle their own
packaging, or app bundle generation, eg. Blazor.
```
But then also there's no exposed way to acccess VFS and a few other features, when it would be convenient to do so.
The documentation in `features.md` is also wrong then, as it lists `AppBundle` as the folder and that is never created from the template, only `wwwroot`.

### Reproduction Steps

1. Install `wasm-tools` and `wasm-experimental` workloads (to have the templates, but that's not strictly required I think)
2. Create a project from the `wasmbrowser` template
3. Enable MSBuild Detailed logging
4. Build or publish the project

### Expected behavior

Target `_WasmGenerateAppBundle` is not skipped.

### Actual behavior

`Target "_WasmGenerateAppBundle" skipped, due to false condition; ('$(WasmGenerateAppBundle)' == 'true') was evaluated as ('false' == 'true').`

### Known Workarounds

Injecting a target that runs as part of the toolchain or at least early enough in the WASM workflow, which sets `WasmGenerateAppBundle` works.

For VFS specifically, I understand there might be some way to provide a custom `MonoConfig` in JS to set a list of assets. I haven't been able to get that working because of errors during the build, which seem to be tied to the MSBuild targets expecting `WasmAppBuilder` to have run (which it doesn't).

### Configuration

.NET 8
Browser WASM

**Comment by RedMike** [comment]
I thought it was weird that the tests hadn't caught this or at least had warnings, it looks like the tests have a warning for this:

```
<Warning
      Condition="'$(WasmMainJS)' != '' and '$(WasmGenerateAppBundle)' != 'true'"
      Text="%24(WasmMainJS) is set when %24(WasmGenerateAppBundle) is not true: it won't be used because an app bundle is not being generated. Possible build authoring error" />
```

But the element isn't `WasmMainJS`, it's `WasmMainJSPath`.

**Comment by maraf** [comment]
The `wasmbrowser` template was migrated to a new SDK that is aligned with Blazor project layout and build.
Some of the properties are not supported in .NET 8. We are working on filling these gaps.

As a work around you have two options
a. You can put files that you need to download to VFS in `wwwroot` and then configure dotnet in JS
```javascript
dotnet.withConfig({ resources: { vfs: { "virtual_path_file_in_vfs.txt": { "relative_path_in_wwwroot_to_file.txt" : null } } } })
```

b. Revert back to previous SDK (it fully works in .NET 8, but will be removed in the future)
You can generate a `wasmconsole` app and override content from `runtimeconfig.template.json`

**Comment by RedMike** [comment]
Right, I was worried it might be something like that. Is there any info on what the plan is for the new SDK for these things (will VFS be used at all, how will we be able to add content to wwwroot from referenced projects, etc), or is it still being decided as things are developed? 

I can build something to work around it with the first option, I just don't want to end up with something that's completely incompatible with how it's meant to work in .NET 9 and later, if there's already info.

**Comment by RedMike** [comment]
Trying the workaround though, I still end up with runtime failures in JS: `Assert failed: Expect to have one dotnetwasm asset in resources`. Because I guess I'm overwriting the config instead of adding to it, that's why I thought I needed a build step that adds it as an asset (so the built blazor.boot.json is still read and used, we just add new elements under `vfs` which doesn't exist there).

I could load `blazor.boot.json` explicitly before `dotnet.create()`, parse it, add the `vfs` entries, and then send that to `withConfig` during the call. It feels a bit brittle but I guess it works.

**Comment by pavelsavara** [comment]
This is the [old functionality](https://github.com/dotnet/runtime/blob/cabacfcdabf1f3fde00edd096dc7412c5e0ec2cf/src/tasks/WasmAppBuilder/WasmAppBuilder.cs#L287)
This is [where it needs to go](https://github.com/dotnet/runtime/blob/main/src/tasks/Microsoft.NET.Sdk.WebAssembly.Pack.Tasks/GenerateWasmBootJson.cs)
when it [arrived from](https://github.com/dotnet/runtime/blob/cabacfcdabf1f3fde00edd096dc7412c5e0ec2cf/src/mono/nuget/Microsoft.NET.Sdk.WebAssembly.Pack/build/Microsoft.NET.Sdk.WebAssembly.Browser.targets#L370) here

the solution with GenerateWasmBootJson should do cache busting hashes for VFS,  PR welcome 😸
